components/src/main/java/org/wso2/carbon/messaging/CloseCarbonMessage.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.messaging;
+
+/**
+ * {@link CarbonMessage} type for Close Messages.
+ */
+public class CloseCarbonMessage extends CarbonMessage {
```

Comment:
But the most generic name I can think of is CloseMessage / CloseCarbonMessage.
It is a carbon message when someone closing the connection 
It can be used in any protocol which provides a status code and reason text (Not necessary).
If we go with control message then it will be a performance overhead for application since it would have to check whether the message is a _ping, pong_ or a _close_.
Then if it is close, then it will have to get the close status code and reason text as properties, cast them and use them.
With the current approach, it is only checking the instance of it and if it is a close message it simply call the _getStatusCode()_ and _getReasonText()_.